src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,45 @@
+import clsx from 'clsx';
+import React from 'react';
+import styles from './Badge.module.css';
+
+export interface Props {
+  /**
+   * Text or icon to be placed on the upper right corner of the badgeable element.
+   *
+   * TODO-JL: prop naming / typing
+   */
+  badge?: React.ReactNode;
```

Comment:
Seeking advice on this prop name and how to type. I'm trying to do something like `badge?: string | Icon` but not sure how I can do that `Icon` part

Reply:
The tricky part is that you have two similarly-named things doing different jobs: this prop `badge` defines what _functions_ as the badge, while `<Badge>` will wrap some other component that will show a badge. Component names should be nouns generally, but in this case `<Badge>` is a verb, and `badge` is the noun.

If you keep `<Badge>`, a prop name that could work is `with`, so the interface is like:

```jsx
<Badge with={<Icon ...>}><SomeComponent /></Badge>
```

There's also the `<Badgeable>` approach with the props corresponding to all available positions. 

A third option would be to rename the component and add subcomponents similar to how our Headless UI -based components work. 

```jsx
<Badge.Wrapper>
    <Badge.Content><SomeComponent /></Badge.Content>
    <Badge.$Type position="top-right">{appropriate content}</Badge.$Type>
</Badge.Wrapper>
```

Where `$Type` is how you control what children are allowed.

All three can work, but curious to hear your thoughts before I mention any of my preferences

Reply:
I do like the `<Badge.Wrapper>` approach, and pushed a commit to represent, but would need more explanation regarding `$Type` or a sync.

Reply:
yah sure, so tl;dr `$Type` wouldn't be literal – I was thinking it would be replaced with stuff like `Text` or `Icon`. But let's chat thru it and make sure that isn't crazy or cumbersome

Reply:
Synced, and to summarize, using a 
`<Badge>` wrapper, and
`<Badge.Icon>` subcomponent or 
`<Badge.Text>` subcomponent